ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -123,6 +132,30 @@ public void run() {
         boolean hasError = false;
         // if some partitions have more requests, continue to handle them.
         while (hasMore && !hasError) {
+          /*
+           * partition compaction control
+           */
+          Set<Long> partitionsUnderRepair = getPartitionIdsNeedRepair();
+          for (long partition : partitionsCompactionDisabled) {
+            if (!partitionsUnderRepair.contains(partition)) {
+              partitionsCompactionDisabled.remove(partition);
+              PartitionId partitionId = clusterMap.getPartitionIdByName(Long.toString(partition));
+              storageManager.controlCompactionForBlobStore(partitionId, true);
+              logger.info("RepairRequests Sender: enable compaction on {} {}", nodeId, partitionId);
+            }
+          }
+          for (long partition : partitionsUnderRepair) {
+            if (!partitionsCompactionDisabled.contains(partition)) {
+              partitionsCompactionDisabled.add(partition);
+              PartitionId partitionId = clusterMap.getPartitionIdByName(Long.toString(partition));
+              storageManager.controlCompactionForBlobStore(partitionId, false);
+              logger.info("RepairRequests Sender: disable compaction on {} {}", nodeId, partitionId);
+            }
+          }
+
+          /*
+           * handle the RepairRequests
+           */
```

Comment:
nit: can we create a new method for this logic?

Reply:
Wrap to one function.